build.js

```diff
@@ -0,0 +1,450 @@
+  formatter: function({dictionary, file, options}) {
+    const {outputReferences} = options
+    return (
+      fileHeader({file}) +
+      `@media (pointer: fine) { :root {\n` +
+      formattedVariables({format: 'css', dictionary, outputReferences}) +
+      '\n}}\n'
+    )
+  }
+})
+
+StyleDictionary.registerFormat({
+  name: 'custom/format/custom-media',
+  formatter(dictionary) {
+    return dictionary.allProperties
+      .map(prop => {
+        const {attributes, value} = prop
+        const size = attributes.type.replace(/_/g, '-') // 1
+        return `@custom-media --${size}-viewport (${value});`
+      })
+      .join('\n')
+  }
+})
+
+// format docs
+StyleDictionary.registerFormat({
+  name: 'json/docs',
+  formatter: function(dictionary, config) {
+    // const obj = dictionary.allProperties.reduce((obj, token) => {
+    //   const {comment, name, filePath} = token
+    //   // const {css, typescript, category} = token.attributes
+    //   console.log('name', filePath)
+    //   // obj[name] = {
+    //   //   formats: {
+    //   //     // css,
+    //   //     // typescript
+    //   //   },
+    //   //   comment
+    //   //   // category
+    //   // }
+    //   obj[name] = {...token}
+    //   return obj
+    // }, {})
+
+    const groupedTokens = _.groupBy(dictionary.allProperties, 'filePath')
+
+    return JSON.stringify(groupedTokens, null, 2)
+  }
+})
+
+// APPLY THE CONFIGURATION
+// IMPORTANT: the registration of custom transforms
+// needs to be done _before_ applying the configuration
+// const StyleDictionaryExtended = StyleDictionary.extend(__dirname + '/config.js')
+
+// FINALLY, BUILD ALL THE PLATFORMS
+// StyleDictionaryExtended.buildAllPlatforms()
+// const arr = location.split('/')
+// const filename = arr[arr.length - 1]
+// const [filenameSansExtension] = filename.split('.')
+
+// build all tokens
+StyleDictionary.extend({
+  source: [`tokens/**/*.json`],
+  platforms: {
+    css: {
+      // transformGroup: `tokens`,
+      buildPath: 'dist/css/',
+      transforms: ['name/css', 'pxToRem'],
+      // map the array of token file paths to style dictionary output files
+      //   files: [{format: 'css/variables', destination: 'test.css'}]
+      files: tokenFiles.map(filePath => {
+        return {
+          format: `css/variables`,
+          destination: filePath.replace(`.json`, `.css`),
+          filter: token => token.filePath === filePath,
+          options: {
+            outputReferences: true
+          }
+        }
+      })
+    },
+    js: {
+      // transformGroup: `tokens`,
+      buildPath: 'dist/js/',
+      transforms: ['name/js/es6', 'pxToRem'],
+      // map the array of token file paths to style dictionary output files
+      files: tokenFiles.map(filePath => {
+        return {
+          format: `javascript/es6`,
+          destination: filePath.replace(`.json`, `.js`),
+          filter: token => token.filePath === filePath
+        }
+      })
+    },
```

Comment:
@rezrah @colebemis can one of you help me declare [ts modules](https://amzn.github.io/style-dictionary/#/formats?id=typescriptmodule-declarations) with my `.map` business going on here?

This format expects two files per export, but I'm assuming we still want those files to match up the naming convention of all other format exports

```
"files": [
        {
          "format": "javascript/module",
          "destination": "colors.js"
        },
        {
          "format": "typescript/module-declarations",
          "destination": "colors.d.ts"
        }
      ]
```

The `tokenFiles.map(filePath =>` is pushing the original JSON file path and name to the export, instead of manually defining each file.